docs/features/pm2-api.md

```diff
@@ -36,22 +34,120 @@ pm2.connect(function(err) {
     instances : 4,                // Optional: Scale your app by 4
     max_memory_restart : '100M'   // Optional: Restart your app if it reaches 100Mo
   }, function(err, apps) {
-    pm2.disconnect();
+    pm2.disconnect();   // Disconnect from PM2
+    if (err) {
+      throw new Error('err');
+    }
   });
 });
 ```
 
-## No daemon
-
-Make sure you killed pm2 before via `pm2 kill`, then to start pm2 programmatically in no deamon mode, you need to pass the `true` as the first argument of `pm2.connect`:
-
-```javascript
-var pm2 = require('pm2');
+## Programmatic API
 
-pm2.connect(true, function(err) {
-  // Do stuff
-});
-```
+`npm install pm2 --save`
+
+**`pm2.connect(errback)`** - Connects to a running pm2 daemon ("God") or launches and daemonizes one. If launched, the pm2 process will continue running after the script exits.  
+**`pm2.connect(noDaemonMode, errback)`**
+
+* `noDaemonMode` - (Default: false) If true is passed for the first argument, pm2 will not be run as a daemon and will die when the script starting it exits. By deafult, pm2 is stays alive after your script exits. Make sure pm2 isn't already running when you start it in no-daemon mode.  
+* `errback(error)` - Called when finished connecting to or launching the pm2 daemon process.
+
+**`pm2.start(options, errback)`** - Starts a script that will be managed by pm2.  
+**`pm2.start(jsonConfigFile, errback)`**  
+**`pm2.start(script, errback)`**  
+**`pm2.start(script, options, errback)`**  
+**`pm2.start(script, jsonConfigFile, errback)`**  
+
+* `script` - The path of the script to run.
+* `jsonConfigFile` - The path to a JSON file that can contain the same options as the `options` parameter.
+* `errback(err,proc)` - An errback called when the `script` has been started. 
```

Comment:
you could call it `callback`, and specify that it'll be called with an error and a process object. This object might help debugging the start issue.